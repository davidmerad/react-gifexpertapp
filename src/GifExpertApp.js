import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ()=>{
    const [categories, setCategories] = useState(['One Punch']);
    // const handleAdd=() =>{
    //     setCategories(cats => [...cats,"HunterxHunter"]);

    // };
    return (<> 
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories} />
        <hr></hr>       
        <ol>
            {
                categories.map(cat => <GifGrid key={cat} category={cat}/>)
            }
        </ol>
    </>);
}

export default GifExpertApp;