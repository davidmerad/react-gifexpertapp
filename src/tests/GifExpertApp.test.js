import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp";

describe('Pruebas con <GifExpertApp />', () => {
    
    test('Debe mostrar correctamente <GifExpertApp /> ', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot(); 
    });
    test('Debe mostrar una lista de categorias ', () => {
        const categories = ['One Punch','Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot(); 
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
})
