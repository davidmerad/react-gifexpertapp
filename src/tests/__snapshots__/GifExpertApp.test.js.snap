// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifExpertApp /> Debe mostrar correctamente <GifExpertApp />  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas con <GifExpertApp /> Debe mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
